spdlog@1.15.0.bcr.2

jwnimmer-tri opened the pull request:
Compared to `1.15.0.bcr.1`, add a `bool_flag` for header_only mode and now depends  on `@bazel_skylib`.

This enables `--@spdlog//:header_only=False` on the bazel command line.

---

See also #1769 which requests a mechanism to enable `USE_STD_FORMAT`.  If desirable, I could try adding that option as an additional flag in this pull request, or in a follow-up change.

bazel-io commented:
Hello @Vertexwahn, @c8ef, modules you maintain (spdlog) have been updated in this PR. Please review the changes.

jwnimmer-tri commented:
@bazel-io skip_check unstable_url

Vertexwahn commented on modules/spdlog/1.15.0.bcr.2/MODULE.bazel:
```diff
@@ -0,0 +1,11 @@
+module(
+    name = "spdlog",
+    version = "1.15.0.bcr.2",
+    compatibility_level = 1,
```
Add `bazel_compatibility = [">=7.2.1"],` since you depend on overlay feature from Bazel

Vertexwahn commented on modules/spdlog/1.15.0.bcr.2/presubmit.yml:
```diff
@@ -0,0 +1,18 @@
+    - ubuntu2204
+    - ubuntu2404
+    - windows
+  bazel: [7.x, rolling]
```
You can change this to `bazel: [7.x, 8.x, rolling]`

Vertexwahn commented:
Can we test also the` --@spdlog//:header_only=False` variant in CI?
`USE_STD_FORMAT` as a flag would also be nice - but we can defer this to a later PR...

jwnimmer-tri committed:
spdlog@1.15.0.bcr.2

Add bool_flag for header_only mode.
This enables `--@spdlog//:header_only=False` on the bazel command line.
Requires a new dependency on @bazel_skylib.

Add bazel_compatibility metadata to ensure that overlay is supported.

Add bazel 8.x to presubmit testing.
Add /utf-8 for windows presubmit testing.

jwnimmer-tri commented:
Good idea; test added.

bazel-io reviewed (dismissed):
Hello @bazelbuild/bcr-maintainers, all modules in this PR have been approved by their maintainers. Please take a final look to merge this PR.

jwnimmer-tri commented:
The new presubmit matrix was failing on windows with UTF-8 complaints.  I added `--cxxopt=/utf-8` to fix it, similar to the existing pattern for `fmt`.

bazel-io reviewed (approved):
Hello @bazelbuild/bcr-maintainers, all modules in this PR have been approved by their maintainers. Please take a final look to merge this PR.
